java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java

```diff
@@ -278,6 +278,11 @@
      */
     public static final String SALT_EVENT_THREAD_POOL_SIZE = "java.salt_event_thread_pool_size";
 
+    /**
+     * Number of threads dedicated to processing Salt events with no minion id.
+     */
+    public static final String SALT_GLOBAL_EVENT_THREAD_POOL_SIZE = "java.salt_global_event_thread_pool_size";
```

Comment:
I appreciate the effort to make this very general, but I would suggest starting with a hardcoded pool size of 1 for simplicity's sake.

Only when we see a real performance bottleneck here I would add more threads - along with proper testing that no two no-minion-id events can conflict when executed on separate threads, or out-of-order.

I would suggest to fix the size to 1 in a separate commit, so that we retain this code the day we decide to add it back.

Probably I will have some more comments once this is in.

Reply:
Done